src/viam/app/data_client.py

```diff
@@ -466,8 +469,8 @@ async def binary_data_capture_upload(
         sensor_contents = SensorData(
             metadata=(
                 SensorMetadata(
-                    time_requested=datetime_to_timestamp(data_request_times[0]) if data_request_times[0] else None,
-                    time_received=datetime_to_timestamp(data_request_times[1]) if data_request_times[1] else None,
+                    time_requested=datetime_to_timestamp(data_request_times[0]) if data_request_times else None,
+                    time_received=datetime_to_timestamp(data_request_times[1]) if data_request_times else None,
```

Comment:
What if `data_request_times[0] and not data_request_times[1]`? Same question below.

Reply:
So `data_request_times` is annotated as an `Optional[Tuple[datetime, datetime]]`, not a `Tuple[Optional[datetime], Optional[datetime]]`, so there should either be a tuple with two `Some`s or a `None`. So I think the case you're describing is one that's inconsistent with how we expect input. Definitely open to discussing changing what we expect though!

Reply:
Ah no I misunderstood the typing; that sounds right to me thanks for the context!